app/controllers/api/v1/characters_controller.rb

```diff
@@ -2,27 +2,35 @@ class Api::V1::CharactersController < Api::ApiController
   before_action :login_required, only: [:update, :reorder]
   before_action :find_character, except: [:index, :reorder]
   before_action :find_post, except: [:update, :reorder]
+  before_action :find_template, only: :index
   before_action :require_permission, only: :update
 
   resource_description do
     description 'Viewing and editing characters'
   end
 
   api :GET, '/characters', 'Load all the characters that match the given query, results ordered by name'
-  param :q, String, required: false, desc: "Query string"
+  param :q, String, required: false, desc: "If provided, will return only characters where q is present as a substring anywhere in any of a character's name, screenname or template nickname."
   param :page, :number, required: false, desc: 'Page in results (25 per page)'
   param :post_id, :number, required: false, desc: 'If provided, will return only characters that appear in the provided post'
+  param :template_id, :number, required: false, desc: 'If provided, will return only characters that belong to the provided template. Use 0 to find only characters with no template.'
+  param :includes, Array, in: ["default", "aliases", "template_name"], of: String, required: false, desc: 'Specify additional fields to return in JSON'
   error 403, "Post is not visible to the user"
   error 422, "Invalid parameters provided"
   def index
-    queryset = Character.with_name(params[:q].to_s).ordered
+    queryset = Character.ordered
+    queryset = queryset.with_name(params[:q].to_s) if params[:q].present?
+    queryset = queryset.where(user_id: current_user.id) if params[:user_id].present?
```

Comment:
Do we need to specify `param :user_id` above? (Or is it inherited somehow? I haven't touched API code in forever.)

Reply:
Oops this looks like a relic of the charpicker, I'll update it yes.